controllers/object_controls.go

```diff
@@ -773,6 +783,48 @@ func applyCommonDaemonsetConfig(obj *appsv1.DaemonSet, config *gpuv1.ClusterPoli
 	return nil
 }
 
+// Transform the host-root volume if required for the Daemonset passed as input
+func transformHostRootVolume(obj *appsv1.DaemonSet, hostRoot string) {
+	if hostRoot == "" || hostRoot == "/" {
+		return
+	}
+
+	containsHostRootVolume := false
+	for index, volume := range obj.Spec.Template.Spec.Volumes {
+		if volume.Name == "host-root" {
+			obj.Spec.Template.Spec.Volumes[index].HostPath.Path = hostRoot
+			containsHostRootVolume = true
+			break
+		}
+	}
+
+	if !containsHostRootVolume {
+		return
+	}
+
+	for index := range obj.Spec.Template.Spec.InitContainers {
+		setContainerEnv(&(obj.Spec.Template.Spec.InitContainers[index]), HostRootEnvName, hostRoot)
+	}
+
+	for index := range obj.Spec.Template.Spec.Containers {
+		setContainerEnv(&(obj.Spec.Template.Spec.Containers[index]), HostRootEnvName, hostRoot)
+	}
+}
+
+// Transform the host-dev-char volume if required for the Daemonset passed as input
+func transformHostDevCharVolume(obj *appsv1.DaemonSet, hostRoot string) {
+	if hostRoot == "" || hostRoot == "/" {
+		return
+	}
```

Comment:
See my comment on adding a single `transformForHostRoot` function where we can do this check once.

Reply:
Done.